src/AgDatabase.cs

```diff
@@ -99,8 +99,15 @@ public bool Exists()
     /// </summary>
     public void Delete()
     {
+      _listener.ForEachAgInstance((s, ag) => {
+        if (!ag.IsPrimaryInstance)
+        {
+          ag.Remove(Name);
+          s.Database(Name)?.Drop();
```

Comment:
as the stackoverflow post mentions, is it worth calling `.SuspendDataMovement();` as well?

Reply:
My gut feeling is yes, but I need to look into further. 
I'll add a more fleshed out rationale for why it is/isn't worthwhile once I've researched a bit more.

Reply:
From what I can tell after some researching, `SuspendDataMovement()` may not be necessary. The process of removing the database from the availability group _should_ stop the synchronization on the secondary's (and I have yet to see any reported errors for attempting to remove an availability database while data is synchronizing). The biggest difference I can see is that suspending HADR will keep the log queues intact, i.e. the primaries send queue will continue to build up logs until the database is removed and the corresponding queue is presumably destroyed.

Reply:
> The process of removing the database from the availability group should stop the synchronization on the secondary's  
yes

> suspending HADR will keep the log queues intact, i.e. the primaries send queue will continue to build up logs until the database is removed and the corresponding queue is presumably destroyed.  
yes, that would make a lot of sense!  Since we are dropping the DB in this case, I think we are good to remove the DB from the AG without worrying about keeping the log queue :)

good to go!